doc/sphinx/using/httpproxy.rst

```diff
@@ -0,0 +1,194 @@
+
+The primary use is for network devices, that happen to have more and more HTTPS APIs in addition to the usual ``conf terminal`` available through SSH. As the same commands are usually available from HTTPS and SSH on these devices, it would be too bad to secure the access to SSH through the bastion, but leave direct access to their HTTPS API!
+
+Query workflow
+==============
+
+The workflow is similar to the one used by SSH, e.g. two distinct connections (ingress and egress), with the egress connection using credentials stored on the bastion:
+
+- A client makes an HTTP request to the proxy, with the following information embedded in:
+
+  - The type of request (GET or POST)
+  - The complete URI, including the host of the remote HTTPS server it would like to send the request to
+  - Potential body data for POST requests
+  - Credentials to authenticate to the proxy on the ingress connection, namely the bastion account name and its proxy password (set by ``selfGenerateProxyPassword``)
+  - User name to use to authenticate on the remote HTTPS server (for the egress connection)
+
+- The bastion checks the provided credentials to authenticate the request against a known account (authentication part)
+- The bastion verifies whether the just-authenticated account has access rights to connect to the remote server as the specified remote user (authorization part)
+- The bastion uses the (group or personal) credentials stored on the bastion, to passthrough the HTTP request to the remote server, as the specified remote user
+- The bastion forwards the response to the client
+
+Setting up the HTTPS Proxy
+==========================
+
+You should enable the HTTPS Proxy daemon, and configure it. Please check the :doc:`/administration/configuration/osh-http-proxy_conf` for more information.
+
+Running a query through the proxy
+=================================
+
+First try
+---------
+
+Once the proxy is running, we can try to query it:
+
+.. code-block:: none
+   :emphasize-lines: 1
+
+   $ curl https://bastion1.example.org:8443/
+   No authentication provided, and authentication is mandatory
+
+Of course, the proxy only accepts to work only when one is properly authenticated to it. To do this, one should have an account on the bastion, and use the :doc:`/plugins/open/selfGenerateProxyPassword` command so that a new ingress password is set for their account. They'll then be able to authenticate to the proxy using the HTTP basic-auth method, and try to send a request to a remote server. To keep a high compatibility with HTTP clients and libraries that can be used on the ingress side, all the additional data required by the bastion to properly authenticate, authorize and passthrough the request is encoded in the *user* part of the widely supported HTTP Authorize header (basic-auth). The *password* part corresponds to the password we've generated just above.
```

Comment:
```suggestion
Of course, the proxy only accepts to work when one is properly authenticated to it. To do this, one should have an account on the bastion, and use the :doc:`/plugins/open/selfGenerateProxyPassword` command so that a new ingress password is set for their account. They'll then be able to authenticate to the proxy using the HTTP basic-auth method, and try to send a request to a remote server. To keep a high compatibility with HTTP clients and libraries that can be used on the ingress side, all the additional data required by the bastion to properly authenticate, authorize and passthrough the request is encoded in the *user* part of the widely supported HTTP Authorize header (basic-auth). The *password* part corresponds to the password we've generated just above.
```